src/tactic/compute_degree.lean

```diff
@@ -130,6 +129,26 @@ meta def eval_guessing (n : ℕ) : expr → tactic ℕ
 | `(max %%a %%b) := max <$> eval_guessing a <*> eval_guessing b
 | e              := eval_expr' ℕ e <|> pure n
 
+/--  A simple check: `check_target_changes t` fails if `t` unifies with one of the current
+goals.  I use it to make sure that the tactics are actually making progress, by feeding the target
+`t`, stored before applying them. -/
+meta def check_target_changes (t : expr) : tactic unit :=
```

Comment:
I think this will be more useful if it wraps the tactic call, something like:
```suggestion
meta def check_target_changes (tac : tactic α) : tactic α :=
do
  t ← target,
  x ← tac,
  gs ← get_goals >>= list.mmap infer_type,
  (success_if_fail $ gs.mfirst $ unify t) <|> fail "Goal did not change",
  pure x
```

Reply:
Should this be moved to `tactic/core.lean`?

Reply:
@Vierkantor, thank you very much for the review!

I moved `check_target_changes` to `tactic.core`.  I also added a `focus1` around it, since there may be two equal goals to begin with and this check is supposed to make sure that the goal to which `tac` is going to apply will not generate the same goal again.